### PR TITLE
Added Currency formatting to TextField

### PR DESCRIPTION
**Pull request by CarloPalinckx**
### This PR:

proposed release: `minor`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)

**Adds** ✨
- currency formatting to `TextField`
- `onBlur` props to `TextField`
- `polyfill` as export which provides users with polyfills for bricks

**Changes** 🌀
- `npm run test` no longer run with the coverage flag, `npm run test-coverage` now does.
-  typescript to `2.8`

### CODE REVIEW

File: src/components/TextField/formatters/withCurrencyFormatting/index.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        private handleChange = (value: string): void => {
+            this.setState({ value: this.parseIn(value) });
+            this.props.onChange(this.parseOut(value));
+        };
+
+        private handleBlur = (): void => {
+            this.setState({ value: this.format(this.state.value) });
+        };
+
+        public componentWillMount(): void {
```
Comment (woutsluiter) [inline src/components/TextField/formatters/withCurrencyFormatting/index.tsx]: 😱

Comment (CarloPalinckx) [reply src/components/TextField/formatters/withCurrencyFormatting/index.tsx]: oops i did a deprecate

File: src/components/TextField/formatters/withCurrencyFormatting/index.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        }
+
+        private handleChange = (value: string): void => {
+            this.setState({ value: this.parseIn(value) });
+            this.props.onChange(this.parseOut(value));
+        };
+
+        private handleBlur = (): void => {
+            this.setState({ value: this.format(this.state.value) });
+        };
```
Comment (hborghols) [inline src/components/TextField/formatters/withCurrencyFormatting/index.tsx]: When one of the settings is changed (e.g., EUR to USD) this only shows after triggering a blur event. This is a bit counter-intuitive when not in a focus state imo. For example if you would enter an amount and then realize you would like to use a different currency, I would expect the currency symbol and formatting immediately after finishing my selection when the input is not focused.

File: src/components/TextField/formatters/withCurrencyFormatting/index.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        }
+
+        private parseIn(value: string): string {
+            const regex = new RegExp(`[^0-9${this.state.decimalSeperator}]`);
+
+            return value.replace(regex, '');
+        }
+
+        private parseOut(value: string): number {
+            const parsed = parseFloat(value.replace(this.state.decimalSeperator, '.'));
```
Comment (hborghols) [inline src/components/TextField/formatters/withCurrencyFormatting/index.tsx]: Large numbers are not parsed correctly after having been parsed once. When parsing numbers > 1000, the combination of decimal seperators and thousands seperators causes rounding errors when triggering another blur event. 

For example: when entering `1360` in EUR, this is formatted as `1.360,00`. When subsequently triggering another blur event without changing the input, this is reformatted to `1,36`.

File: src/components/TextField/formatters/withCurrencyFormatting/test.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                value: '19000000.121111111',
+            },
+        });
+
+        component.find('input').simulate('blur');
+
+        expect(component.find('input').prop('value')).toEqual('19,000,000.12');
+        expect(component.find(TextField).prop('prefix')).toEqual('$');
+        expect(changeMock).toHaveBeenCalledWith(19000000.12);
+    });
```
Comment (hborghols) [inline src/components/TextField/formatters/withCurrencyFormatting/test.tsx]: Maybe a test should be added to make sure input is not formatted multiple times with different results (i.e., the rounding errors case described above).

**Comment by hborghols** [comment]
The field does not show prefix or suffix in safari and formatting doesn't work. Disallowed input (e.g., `aaaaaa`) is prevented correctly but no formatted output is shown.

**Comment by CarloPalinckx** [comment]
> The field does not show prefix or suffix in safari, IE11 and Edge, and formatting doesn't work. 

This is correct. Bricks will now require a polyfill to run. I figured this could be a separate issue because we could provide an easy way to provide other polyfills as well. Will add an issue 👍

**Comment by CarloPalinckx** [comment]
Done ✅ , #160

**Pull request merged**